src/main/webapp/scripts/post-comment.js

```diff
@@ -0,0 +1,27 @@
+function postVideoComment(videoId, commentText) {
+    return new Promise((resolve, reject) => {
+        gapi.client.request(buildRequestBody(videoId, commentText))
+            .then(res => res.json)
+            .then(data => resolve(data))
+            .catch(err => reject(err.result.error.message));
```

Comment:
Could you instead reject a custom error object with the error message? Strings aren't consider errors.

Reply:
lol fair point I will use reject(Error(...))

Reply:
Done